### PR TITLE
1225 - Test Framework Switch to Playwright

### LINKED ISSUE #1225: Testing: Switch Test Framework (Cypress or Playwright)
**Describe the bug**

Do a full POC of cypress to make sure it can replace our current system.

**Needs**

- [ ] remove unused deps
- [ ] example tests for a few components
- [ ] e2e tests with coverage
- [ ] component (functional) tests with coverage
- [ ] works in real browser(s) - chrome for now
- [ ] docs on how to use and convert old tests
- [ ] percy tests with new version / or image tests
- [ ] runs on ci git
- [ ] fast/faster
- [ ] a memory leak test
- [ ] axe test
- [ ] make sure no costs
- [ ] use all TS / Es modules ect
- [ ] remove all test code except uncovered tests

**Additional context**

### PR DESCRIPTION
**Pull request by tmcconechy**
**Explain the details for making this change. What existing problem does the pull request solve?**

Switches out the test framework to https://playwright.dev/. 

⚠️ NOTE: As of now on this PR I'm working on the CI to make sure it works. Not 100% complete ⚠️ 

**Related github/jira issue (required)**:
Fixes #1225

**Steps necessary to review your pull request (required)**:

- See Testing.md
- Try the commands:

```
npm run test:coverage
npm run test:coverage:no-build
npm run test
npx playwright test -- tag
```
- CI works ok and shows coverage.
- tests in tests-to-convert need converting 
- install plugin https://marketplace.visualstudio.com/items?itemName=ms-playwright.playwright 
- then click the "beaker" icon in visual code try and run the tests this way.. can review the plugin docs

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Review by EdwardCoyle — APPROVED** [review]
Amazing work @tmcconechy!  Thank you for your efforts on this one

**Review by EdwardCoyle — DISMISSED** [review]

**Comment by ericangeles** [comment]
Fantastic work, @tmcconechy! 🍌 By any chance, does the UI mode work on this?

**Comment by ericangeles** [comment]
Confirming that the UI mode is working as well! This is awesome!

Screenshot 2023-11-20 at 11 07 34 PM

File: package.json [hunk 1]
```diff
@@ -32,83 +32,77 @@
...
-    "test:percy": "npx jest --verbose --runInBand --detectOpenHandles --logHeapUsage --forceExit -- percy",
-    "test:percy:local": "npx percy exec -- npm run test:percy",
-    "test:debug": "PERCY_LOGLEVEL=silent node --inspect-brk node_modules/.bin/jest --runInBand --detectOpenHandles --forceExit",
+    "test": "PERCY_LOGLEVEL=silent npx playwright test",
+    "test:percy": "npx percy exec -- npm run test",
+    "test:coverage": "npx rimraf .nyc_output && npm run build:coverage && npm run test && npx nyc report && open coverage/index.html",
+    "test:coverage:no-build": "npx rimraf .nyc_output && npm run test && npx nyc report && open coverage/index.html",
+    "test:ci": "PERCY_LOGLEVEL=warn npx playwright test && npx nyc report",
+    "test:update-snapshots": "PERCY_LOGLEVEL=silent npx playwright test --update-snapshots",
+    "test:ui": "PERCY_LOGLEVEL=silent npx playwright test --uic",
```
Comment (ericangeles) [inline package.json:47]: You probably mispelled it, @tmcconechy. `--uic` should be `--ui`.

**Review by ericangeles — APPROVED** [review]

**Comment by tmcconechy** [comment]
Good catch @ericangeles fixed that. But i found that the VIsual Studio plugin does mostly the same thing (except the playback options). So if you didnt definately try the plugin as its real handy

**Review by ericangeles — APPROVED** [review]

**Review by EdwardCoyle — APPROVED** [review]

**Pull request merged**